File: src/main/java/net/artux/pdanetwork/service/items/SellerService.java
package net.artux.pdanetwork.service.items;

import net.artux.pdanetwork.models.Status;
import net.artux.pdanetwork.models.seller.SellerAdminDto;
import net.artux.pdanetwork.models.seller.SellerDto;

import java.util.List;
import java.util.UUID;

public interface SellerService {

    SellerDto getSeller(long id);

    List<SellerAdminDto> getSellers();

    Status buy(long sellerId, UUID id, int quantity);

    Status sell(long sellerId, UUID id, int quantity);

    SellerDto createSeller(SellerAdminDto dto);

    SellerDto updateSeller(Long id, SellerAdminDto dto);

    SellerDto addSellerItems(Long sellerId, List<String> s);

    SellerDto deleteSellerItems(Long sellerId, List<UUID> ids);

    void deleteSeller(Long id);

}

File: pdanet/src/main/java/net/artux/pdanetwork/service/items/SellerService.kt
package net.artux.pdanetwork.service.items

import net.artux.pdanetwork.models.Status
import net.artux.pdanetwork.models.seller.SellerAdminDto
import net.artux.pdanetwork.models.seller.SellerDto
import java.util.UUID

interface SellerService {
    fun getSeller(id: Long): SellerDto

    fun getSellers(): List<SellerAdminDto>

    fun buy(sellerId: Long, id: UUID, quantity: Int): Status

    fun sell(sellerId: Long, id: UUID, quantity: Int): Status

    fun createSeller(dto: SellerAdminDto): SellerDto

    fun updateSeller(id: Long, dto: SellerAdminDto): SellerDto

    fun addSellerItems(sellerId: Long, s: List<String>): SellerDto

    fun deleteSellerItems(sellerId: Long, ids: List<UUID>): SellerDto

    fun deleteSeller(id: Long)

    fun restoreSellersItems()
}
